src/main/java/frc/robot/IMU/BNO055.java

```diff
@@ -0,0 +1,776 @@
+		data.gyro = (byte) ((rawCalData >> 4) & 0x03);
+		data.accel = (byte) ((rawCalData >> 2) & 0x03);
+		data.mag = (byte) (rawCalData & 0x03);
+
+		return data;
+	}
+
+	/**
+	 * Returns true if all required sensors (accelerometer, magnetometer,
+	 *   gyroscope) have completed their respective calibration sequence.
+	 *   Only sensors required by the current operating mode are checked.
+	 *   See Section 3.3.
+	 * @return true if calibration is complete for all sensors required for the
+	 *   mode the sensor is currently operating in. 
+	 */
+	public boolean isCalibrated() {
+		boolean retVal = true;
+		
+		//Per Table 3-3
+		boolean[][] sensorModeMap = new boolean[][]{
+			//{accel, mag, gyro}
+			{false, false, false}, // OPERATION_MODE_CONFIG
+			{ true, false, false}, // OPERATION_MODE_ACCONLY
+			{false,  true, false}, // OPERATION_MODE_MAGONLY
+			{false, false,  true}, // OPERATION_MODE_GYRONLY
+			{ true,  true, false}, // OPERATION_MODE_ACCMAG
+			{ true, false,  true}, // OPERATION_MODE_ACCGYRO
+			{false,  true,  true}, // OPERATION_MODE_MAGGYRO
+			{ true,  true,  true}, // OPERATION_MODE_AMG
+			{ true, false,  true}, // OPERATION_MODE_IMUPLUS
+			{ true,  true, false}, // OPERATION_MODE_COMPASS
+			{ true,  true, false}, // OPERATION_MODE_M4G
+			{ true,  true,  true}, // OPERATION_MODE_NDOF_FMC_OFF
+			{ true,  true,  true}  // OPERATION_MODE_NDOF
+		};
+
+		CalData data = getCalibration();
+		
+		if(sensorModeMap[_mode][0]) //Accelerometer used
+			retVal = retVal && (data.accel >= 3);
+		if(sensorModeMap[_mode][1]) //Magnetometer used
+			retVal = retVal && (data.mag >= 3);
+		if(sensorModeMap[_mode][2]) //Gyroscope used
+			retVal = retVal && (data.gyro >= 3);
+		
+		return retVal;
+	}
+	
+	/**
+	 * Get the sensors internal temperature.
+	 * @return temperature in degrees celsius.
+	 */
+	public int getTemp() {
+		return (read8(reg_t.BNO055_TEMP_ADDR));
+	}
+
+	/**
+	 * Gets a vector representing the sensors position (heading, roll, pitch).
+	 * heading:    0 to 360 degrees
+	 * roll:     -90 to +90 degrees
+	 * pitch:   -180 to +180 degrees
+	 *
+	 * For continuous rotation heading (doesn't roll over between 360/0) see
+	 *   the getHeading() method.
+	 *
+	 * Maximum data output rates for Fusion modes - See 3.6.3
+	 * 
+	 * Operating Mode		Data Output Rate
+	 *   IMU                  100 Hz
+	 *   COMPASS               20 Hz
+	 *   M4G                   50 Hz
+	 *   NDOF_FMC_OFF         100 Hz
+	 *   NDOF                 100 Hz
+	 *
+	 * @return a vector [heading, roll, pitch]
+	 */
+	public double[] getVector() {
+		return xyz;
+	}
+	
+	/**
+	 * The heading of the sensor (x axis) in continuous format. Eg rotating the
+	 *   sensor clockwise two full rotations will return a value of 720 degrees.
+	 * The getVector method will return heading in a constrained 0 - 360 deg
+	 *   format if required.
+	 * @return heading in degrees
+	 */
+	public double getHeading() {
+		return xyz[0] + turns * 360;
```

Comment:
This is going to be relative to when this obj was constructed & the IMU initialized